File: SearchingAlgorithms/LinearSearch/LinearSearch.js
export class LinearSearch {
  find(source, target) {
    if (Array.isArray(source)) {
      for (let i = 0; i < source.length; i++) {
        if (source[i] == target) return i;
      }
    }

    if (typeof source == "string") {
      for (let i = 0; i < source.length; i++) {
        if (source.charAt(i) == target) {
          return i;
        }
      }
    }

    return -1;
  }

  has(source, target) {
    if (Array.isArray(source)) {
      for (let i = 0; i < source.length; i++) {
        if (source[i] == target) return true;
      }
    }

    if (typeof source == "string") {
      for (let i = 0; i < source.length; i++) {
        if (source.charAt(i) == target) {
          return true;
        }
      }
    }
    return false;
  }

  findIn2DArr(arr, target) {
    for (let row = 0; row < arr.length; row++) {
      for (let col = 0; col < arr[row].length; col++) {
        if (arr[row][col] == target) {
          return [row, col];
        }
      }
    }
    return [-1, -1];
  }
  hasIn2DArr(arr, target) {
    for (let row = 0; row < arr.length; row++) {
      for (let col = 0; col < arr[row].length; col++) {
        if (arr[row][col] == target) {
          return true;
        }
      }
    }
    return false;
  }
}

File: SearchingAlgorithms/LinearSearch/LinearSearch.java
package com.searchAlgos;

public class LinearSearch {

	public static int find(int[] arr, int target) {

		for (int i = 0; i < arr.length; i++) {
			if (arr[i] == target) {
				return i;
			}
		}

		return -1;

	}

	public static boolean has(int[] arr, int target) {

		for (int i = 0; i < arr.length; i++) {
			if (arr[i] == target) {
				return true;
			}
		}

		return false;

	}

	public static int find(String word, char f) {

		for (int i = 0; i < word.length(); i++) {
			if (word.charAt(i) == f)
				return i;

		}

		return -1;
	}

	public static boolean has(String word, char f) {

		for (int i = 0; i < word.length(); i++) {
			if (word.charAt(i) == f)
				return true;

		}

		return false;
	}

	public static int[] find(int[][] twoDArr, int target) {

		for (int row = 0; row < twoDArr.length; row++) {
			for (int col = 0; col < twoDArr[row].length; col++) {

				if (twoDArr[row][col] == target) {
					return new int[] { row, col };
				}

			}

		}

		return new int[] { -1, -1 };
	}
	
	public static boolean has(int[][] twoDArr, int target) {

		for (int row = 0; row < twoDArr.length; row++) {
			for (int col = 0; col < twoDArr[row].length; col++) {

				if (twoDArr[row][col] == target) {
					return true;
				}

			}

		}

		return false;
	}
}
